space/authz/authz.go

```diff
@@ -18,97 +22,108 @@ import (
-func (m *KeycloakAuthzServiceManager) EntitlementEndpoint() string {
-	return m.entitlementEndpoint
-}
-
 // KeycloakAuthzService implements AuthzService interface
 type KeycloakAuthzService struct {
-	config AuthzConfiguration
+	config auth.ServiceConfiguration
+	doer   rest.HttpDoer
 }
 
 // NewAuthzService constructs a new KeycloakAuthzService
-func NewAuthzService(config AuthzConfiguration) *KeycloakAuthzService {
-	return &KeycloakAuthzService{config: config}
+func NewAuthzService(config auth.ServiceConfiguration) *KeycloakAuthzService {
+	return &KeycloakAuthzService{config: config, doer: rest.DefaultHttpDoer()}
 }
 
 // Configuration returns authz service configuration
-func (s *KeycloakAuthzService) Configuration() AuthzConfiguration {
+func (s *KeycloakAuthzService) Configuration() auth.ServiceConfiguration {
 	return s.config
 }
 
-// Authorize returns true and the corresponding Requesting Party Token if the current user is among the space collaborators
-func (s *KeycloakAuthzService) Authorize(ctx context.Context, entitlementEndpoint string, spaceID string) (bool, error) {
+// Authorize returns true if the current user is among the space collaborators
+func (s *KeycloakAuthzService) Authorize(ctx context.Context, spaceID string) (bool, error) {
 	jwttoken := goajwt.ContextJWT(ctx)
 	if jwttoken == nil {
 		return false, errors.NewUnauthorizedError("missing token")
 	}
-	return s.checkEntitlementForSpace(ctx, *jwttoken, entitlementEndpoint, spaceID)
+	return s.checkRole(ctx, *jwttoken, spaceID)
 }
 
-func (s *KeycloakAuthzService) checkEntitlementForSpace(ctx context.Context, token jwt.Token, entitlementEndpoint string, spaceID string) (bool, error) {
+type Roles struct {
+	Data []Role `json:"data"`
+}
+
+type Role struct {
+	RoleName   string `json:"role_name"`
+	AssigneeID string `json:"assignee_id"`
+}
+
+func (s *KeycloakAuthzService) checkRole(ctx context.Context, token jwt.Token, spaceID string) (bool, error) {
 	if !s.config.IsAuthorizationEnabled() {
 		// Keycloak authorization is disabled by default in Developer Mode
 		log.Warn(ctx, map[string]interface{}{
 			"space_id": spaceID,
 		}, "Authorization is disabled. All users are allowed to operate the space")
 		return true, nil
 	}
-	resource := auth.EntitlementResource{
-		Permissions: []auth.ResourceSet{{Name: spaceID}},
+	currentIdentityID, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return false, err
 	}
-	ent, err := auth.GetEntitlement(ctx, entitlementEndpoint, &resource, token.Raw)
+
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/resources/%s/roles", s.config.GetAuthServiceURL(), spaceID), nil)
 	if err != nil {
 		return false, err
 	}
-	return ent != nil, nil
+	req.Header.Set("Authorization", "Bearer "+token.Raw)
+	res, err := s.doer.Do(ctx, req)
+	if err != nil {
+		return false, errors.NewInternalError(ctx, err)
+	}
+	defer rest.CloseResponse(res)
+	bodyString := rest.ReadBody(res.Body)
+	if res.StatusCode != http.StatusOK {
```

Comment:
We are assuming that 
- there would always be at least one assignee ( creator/admin )
- the space resource should always exist. Hence, a getting a 404 would be really odd.

which I think are fair assumptions.